### PR TITLE
Week 6 Ex 1 - 4 wip

### PR DESCRIPTION
**Pull request by tienwei**
Sorry for the late PR. Any feedback will be greatly appreciated. Thanks!

### CODE REVIEW

File: cis194/week6/tien/src/Fibonacci.hs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+module Fibonacci where
+
+fib :: Integer -> Integer
+fib 0 = 0
+fib 1 = 1
+fib n = fib (n-1) + fib (n-2)
+
+-- exercise 1 --
+fibs1::[Integer]
+fibs1 = map fib [0..]
```
Comment (sinogermany) [inline cis194/week6/tien/src/Fibonacci.hs]: The use of `fmap` or the infix version `<$>` is preferred as we do not need to change the implementation if we change `[Integer]` to any `Functor f => f Integer`

```suggestion
fibs1 = fmap fib [0..]
```

or

```suggestion
fibs1 = fib <$> [0..]
```

File: cis194/week6/tien/src/Fibonacci.hs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+-- exercise 2 --
+fibs2::[Integer]
+fibs2 = 0: 1: next fibs2 where
+  next (x:y:xs) = (x+y) : next(y:xs)
```
Comment (sinogermany) [inline cis194/week6/tien/src/Fibonacci.hs]: I remember @shineli1984 mentioned a while ago that we tend to put a space before and after the symbols in Haskell.

```suggestion
  next (x : y : xs) = (x + y) : next(y : xs)
```

Comment (tienwei) [reply cis194/week6/tien/src/Fibonacci.hs]: My apology....should have run the formatter before commits.

File: cis194/week6/tien/src/Fibonacci.hs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+streamToList :: Stream a -> [a]
+streamToList (Con x xs) = x : (streamToList xs)
+
+instance Show a => Show (Stream a) where
+  show = show . take 20 . streamToList
+
+-- exercise 4 --
+streamRepeat :: a -> Stream a
+streamRepeat x = Con x (streamRepeat x)
```
[the remark this answers is not in the document]
Comment (tienwei) [reply cis194/week6/tien/src/Fibonacci.hs]: @sinogermany any good tips to understand when to apply `applicative`? I am not very familiar with it yet...Thanks!

File: cis194/week6/tien/src/Fibonacci.hs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+streamMap :: (a -> b) -> Stream a -> Stream b
+streamMap f (Con x xs) = Con (f x) (streamMap f xs)
```
[the remark this answers is not in the document]
Comment (tienwei) [reply cis194/week6/tien/src/Fibonacci.hs]: Nice spot!

File: cis194/week6/tien/src/Fibonacci.hs (hunk 3 above)
[the remark this answers is not in the document]
Comment (sinogermany) [reply cis194/week6/tien/src/Fibonacci.hs]: > @sinogermany any good tips to understand when to apply `applicative`? I am not very familiar with it yet...Thanks!

```haskell
f (g a) (h a) === f <$> g <*> h $ a
f a (g a)     === f <*> g $ a
```

[the remark this answers is not in the document]
Comment (tienwei) [reply cis194/week6/tien/src/Fibonacci.hs]: Thanks for the tips! Will try to memorise for now lol. Cheers.

**Pull request merged**